### PR TITLE
QOL updates to APIRule v2alpha1 integration tests

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Remove prefix initialiser, switching to universal initialisers
- Fix generation of namespace name to use a singular instance of RNG, fixing potential namespace conflicts
- Add a separate optional pre-start hook that scales down oathkeeper, masively speeding up APIGateway startup time
- Do not teardown when the test fail, allowing for post-run troubleshooting 

**Pre-Merge Checklist**

- [ ] As a PR reviewer, verify code coverage and evaluate if it is acceptable.

**Related issues**

### CODE REVIEW

File: tests/integration/pkg/hooks/api-gateway.go [hunk 1]
```diff
@@ -114,7 +137,7 @@ var ApplyAndVerifyApiGatewayCrSuiteHook = func() error {
 		}
 
 		return nil
-	}, testcontext.GetRetryOpts()...)
+	}, retry.DelayType(retry.FixedDelay), retry.Delay(500), retry.Attempts(1200))
```
Comment (triffer) [inline tests/integration/pkg/hooks/api-gateway.go:140]: This seems like a very long time to retry the scale down. Why did we chose to make it that long?

File: tests/integration/pkg/hooks/api-gateway.go [hunk 2]
```diff
@@ -69,7 +70,7 @@ var ApiGatewayCrTearDownScenarioHook = func(ctx context.Context, sc *godog.Scena
 	return ctx, nil
 }
 
-var ApplyAndVerifyApiGatewayCrSuiteHook = func() error {
+func ApplyAndVerifyApiGateway(scaleDownOathkeeper bool) error {
```
Comment (triffer) [inline tests/integration/pkg/hooks/api-gateway.go:73]: I think for a an easier to understand API it might be better to not export this function.

**Review by triffer — DISMISSED** [review]
Very good changes, thanks for taking the time to clean this up. 👍

File: tests/integration/pkg/hooks/api-gateway.go (hunk 1 above)
Comment (barchw) [reply tests/integration/pkg/hooks/api-gateway.go:140]: I will revert that to the old value

**Review by triffer — APPROVED** [review]

**Pull request merged**